examples/keras_io/tensorflow/nlp/ner_transformers.py

```diff
@@ -0,0 +1,341 @@
+
+"""
+We will export this data to a tab-separated file format which will be easy to read as a
+`tf.data.Dataset` object.
+"""
+
+
+def export_to_file(export_file_path, data):
+    with open(export_file_path, "w") as f:
+        for record in data:
+            ner_tags = record["ner_tags"]
+            tokens = record["tokens"]
+            if len(tokens) > 0:
+                f.write(
+                    str(len(tokens))
+                    + "\t"
+                    + "\t".join(tokens)
+                    + "\t"
+                    + "\t".join(map(str, ner_tags))
+                    + "\n"
+                )
+
+
+os.mkdir("data")
+export_to_file("./data/conll_train.txt", conll_data["train"])
+export_to_file("./data/conll_val.txt", conll_data["validation"])
+
+"""
+## Make the NER label lookup table
+
+NER labels are usually provided in IOB, IOB2 or IOBES formats. Checkout this link for
+more information:
+[Wikipedia](https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging))
+
+Note that we start our label numbering from 1 since 0 will be reserved for padding. We
+have a total of 10 labels: 9 from the NER dataset and one for padding.
+"""
+
+
+def make_tag_lookup_table():
+    iob_labels = ["B", "I"]
+    ner_labels = ["PER", "ORG", "LOC", "MISC"]
+    all_labels = [(label1, label2) for label2 in ner_labels for label1 in iob_labels]
+    all_labels = ["-".join([a, b]) for a, b in all_labels]
+    all_labels = ["[PAD]", "O"] + all_labels
+    return dict(zip(range(0, len(all_labels) + 1), all_labels))
+
+
+mapping = make_tag_lookup_table()
+print(mapping)
+
+"""
+Get a list of all tokens in the training dataset. This will be used to create the
+vocabulary.
+"""
+
+all_tokens = sum(conll_data["train"]["tokens"], [])
+all_tokens_array = np.array(list(map(str.lower, all_tokens)))
+
+counter = Counter(all_tokens_array)
+print(len(counter))
+
+num_tags = len(mapping)
+vocab_size = 20000
+
+# We only take (vocab_size - 2) most commons words from the training data since
+# the `StringLookup` class uses 2 additional tokens - one denoting an unknown
+# token and another one denoting a masking token
+vocabulary = [token for token, count in counter.most_common(vocab_size - 2)]
+
+# The StringLook class will convert tokens to token IDs
+lookup_layer = keras.layers.StringLookup(vocabulary=vocabulary)
+
+"""
+Create 2 new `Dataset` objects from the training and validation data
+"""
+
+train_data = tf.data.TextLineDataset("./data/conll_train.txt")
+val_data = tf.data.TextLineDataset("./data/conll_val.txt")
+
+"""
+Print out one line to make sure it looks good. The first record in the line is the number of tokens. 
+After that we will have all the tokens followed by all the ner tags.
+"""
+
+print(list(train_data.take(1).as_numpy_iterator()))
+
+"""
+We will be using the following map function to transform the data in the dataset:
+"""
+
+
+def map_record_to_training_data(record):
+    record = tf.strings.split(record, sep="\t")
```

Comment:
Instead of importing `tf`, do `from tensorflow import data as tf_data` and `from tensorflow import string as tf_string` and add a comment that this is for data preprocessing